tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspTool/TspTool.cs

```diff
@@ -0,0 +1,230 @@
+                    pathToSwaggerReadme, outputDirectory, isAzureResourceManagement, fullyCompatible);
+
+                // Validate pathToSwaggerReadme
+                string? readmeValidationResult = ValidateSwaggerReadme(pathToSwaggerReadme);
+                if (readmeValidationResult != null)
+                {
+                    SetFailure();
+                    return new TspToolResponse
+                    {
+                        ErrorMessage = readmeValidationResult
+                    };
+                }
+                // fullPathToSwaggerReadme is not null or empty at this point - already validated
+                var fullPathToSwaggerReadme = Path.GetFullPath(pathToSwaggerReadme.Trim());
+
+                // validate outputDirectory using the extracted method
+                var validationResult = ValidateOutputDirectory(outputDirectory);
+                if (validationResult != null)
+                {
+                    SetFailure();
+                    return new TspToolResponse
+                    {
+                        ErrorMessage = validationResult
+                    };
+                }
+
+                var fullOutputDir = Path.GetFullPath(outputDirectory.Trim());
+                return RunTspClient(fullPathToSwaggerReadme, fullOutputDir, isAzureResourceManagement ?? false, fullyCompatible ?? false);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error occurred while converting swagger to TypeSpec: {pathToSwaggerReadme}, {outputDirectory}", pathToSwaggerReadme, outputDirectory);
+                SetFailure();
+                return new TspToolResponse
+                {
+                    ErrorMessage = $"Failed: An error occurred trying to convert '{pathToSwaggerReadme}': {ex.Message}"
+                };
+            }
+        }
+
+        private static string? ValidateSwaggerReadme(string pathToSwaggerReadme)
+        {
+            if (string.IsNullOrWhiteSpace(pathToSwaggerReadme) || !pathToSwaggerReadme.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
+            {
+                return $"Failed: Invalid pathToSwaggerReadme '{pathToSwaggerReadme}' - must be a non-empty path to a swagger README.md file";
+            }
+
+            var fullPath = Path.GetFullPath(pathToSwaggerReadme.Trim());
+            if (!File.Exists(fullPath))
+            {
+                return $"Failed: pathToSwaggerReadme '{fullPath}' does not exist.";
+            }
+
+            var fullPathToSwaggerReadme = Path.GetFullPath(pathToSwaggerReadme.Trim());
+            if (!File.Exists(fullPathToSwaggerReadme))
+            {
+
+                return $"Failed: pathToSwaggerReadme '{fullPathToSwaggerReadme}' does not exist.";
+            }
+
+            return null; // Validation passed
+        }
+
+        private static string? ValidateOutputDirectory(string outputDir)
```

Comment:
We could probably move these into a `FileHelper` if you're willing.

Reply:
Created a `FileHelper` with `ValidateEmptyDirectory`